src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,483 @@
+            }
+            lineNumber++;
+        }
+
+        FileWriter dukeFileWriter = new FileWriter("data/duke.txt");
+        dukeFileWriter.write(taskDescriptions.toString());
+        dukeFileWriter.close();
+    }
+
+    public static String updateMarkStatus(boolean isMarked, String currentLine) {
+        String markStatus = isMarked ? "1" : "0";
+        return currentLine.substring(0, 4) + markStatus + currentLine.substring(5);
+    }
+
+    public static void addTaskToDukeTextFile(String addedTaskDescription) throws IOException {
+        File dukeFile = new File("data/duke.txt");
+        Scanner dukeFileScanner = new Scanner(dukeFile);
+        String newDukeTextFileContent = copyFileContent(dukeFileScanner);
+        newDukeTextFileContent += addedTaskDescription + System.lineSeparator();
+        FileWriter dukeFileWriter = new FileWriter("data/duke.txt");
+        dukeFileWriter.write(newDukeTextFileContent);
+        dukeFileWriter.close();
+    }
+
+    public static String copyFileContent(Scanner dukeFileScanner) {
+        StringBuilder taskDescriptions = new StringBuilder();
+        while (dukeFileScanner.hasNext()) {
+            String currentLine = dukeFileScanner.nextLine();
+            taskDescriptions.append(currentLine).append(System.lineSeparator());
+        }
+        return taskDescriptions.toString();
+    }
+
+    public static void main(String[] args) {
+        //Attempt to find 'data' directory; Directory created if not found
+        File dataFile = new File("data");
+        try {
+            findDataFile(dataFile);
+        } catch (SecurityException e) {
+            showSecurityExceptionMessage();
+        }
+
+        //Attempt to find 'duke.txt' text file; Text file created if not found
+        File dukeFile = new File("data/duke.txt");
+        try {
+            findDukeTextFile(dukeFile);
+        } catch (IOException e) {
+            showIOExceptionMessage();
+        }
+
+        ArrayList<Task> taskList = new ArrayList<>();
+        int maxTaskIndex = 0;
+
+        //Load task descriptions (if any) into list
+        try {
+            maxTaskIndex = loadDukeTextFile(taskList ,dukeFile);
+        } catch (FileNotFoundException e) {
+            showDukeTextFileNotFoundMessage();
+        }
+
+        //Greet
+        showWelcomeMessage();
+
+        //Read Input
+        Scanner in = new Scanner(System.in);    //create object that reads input
+        String command = in.next();             //variable to store line (input)
+
+        while (!command.equals("bye")) {
+            switch (command) {
+            case "list":
+                printList(taskList, maxTaskIndex);
+                break;
+            case "unmark":
+                markAsUndone(taskList, in);
+                break;
+            case "mark":
+                markAsDone(taskList, in);
+                break;
+            //Add Task
+            case "todo":
```

Comment:
Avoid deep nesting and arrowhead style code. Abstraction would solve this too, try to refactor into methods wherever you see fit.
Code Quality Guideline: Avoid Deep Nesting